sdk/storage/storage-blob-changefeed/src/LazyLoadingBlobStream.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Readable, ReadableOptions } from "stream";
+import { BlobClient, CommonOptions } from "@azure/storage-blob";
+import { AbortSignalLike } from "@azure/core-http";
+import { createSpan } from "./utils/tracing";
+import { CanonicalCode } from "@opentelemetry/api";
+
+/**
+ * Options to configure the LazyLoadingBlobStream.
+ */
+export interface LazyLoadingBlobStreamOptions extends ReadableOptions, CommonOptions {
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
+   *
+   * @type {AbortSignalLike}
+   * @memberof LazyLoadingBlobStreamOptions
+   */
+  abortSignal?: AbortSignalLike;
+}
+
+interface LazyLoadingBlobStreamDownloadBlockOptions extends CommonOptions {
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
+   *
+   * @type {AbortSignalLike}
+   * @memberof LazyLoadingBlobStreamDownloadBlockOptions
+   */
+  abortSignal?: AbortSignalLike;
+}
+
+/**
+ * This class generates a readable stream from a blobClient's data.
+ *
+ * @export
+ * @class LazyLoadingBlobStream
+ */
+export class LazyLoadingBlobStream extends Readable {
+  /**
+   * BlobClient to make download calls with.
+   *
+   * @private
+   * @type {BlobClient}
+   * @memberof LazyLoadingBlobStream
+   */
+  private readonly _blobClient: BlobClient;
+
+  /**
+   * The offset within the blob of the next block we will download.
+   *
+   * @private
+   * @type {number}
+   * @memberof LazyLoadingBlobStream
+   */
+  private _offset: number;
```

Comment:
>_offset [](start = 10, length = 7)

In long time ago without TS, we use "_" as prefix for JS private members. But TS has good support for member markers like private/public. We don't need "_", which is align with other other classes in storage package.

Reply:
Fixed.